sdk/core/azure-core/test/ut/curl_connection_pool_test.cpp

```diff
@@ -601,6 +601,30 @@ namespace Azure { namespace Core { namespace Test {
       }
     }
 
+    TEST(CurlConnectionPool, forceConnectionClosed)
+    {
+      Azure::Core::Http::Request req(
+          Azure::Core::Http::HttpMethod::Get,
+          Azure::Core::Url(AzureSdkHttpbinServer::Status(101)));
+
+      Azure::Core::Http::CurlTransportOptions options;
+      auto connection
+          = CurlConnectionPool::g_curlConnectionPool.ExtractOrCreateCurlConnection(req, options);
```

Comment:
There is a way we can test the internals of the transport.

We can define "interfaces" (pure virtual base classes) for the underlying socket implementation (ie a CURL interface and a WinHTTP interface) and then mock that interface in our test collateral. We can then call into the transport with the mocked CURL/WinHTTP and have the CURL/WinHTTP behave in whatever fashion we want.

It's a lot of work, but it is completely possible (it will be rather challenging to do this for WinHTTP since we're now using WinHTTP asynchronously, but it is absolutely possible).

Reply:
@ahsonkhan, I am going to resolve this comment thread now so that I can merge this PR.
If you'd like more follow up your question, please create a work item so that we can follow up on it later.